packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+      delegatee1,
+      FixidityLib.newFixedFraction(percentToDelegate, 100).unwrap()
+    );
+  }
+
+  function test_ShouldRevert_WhenDelegatingAsValidator() public {
+    validators.setValidator(delegator);
+    vm.startPrank(delegator);
+    vm.expectRevert("Validators cannot delegate votes.");
+    lockedGold.delegateGovernanceVotes(delegatee1, FixidityLib.newFixedFraction(1, 1).unwrap());
+  }
+
+  function test_ShouldRevert_WhenDelegatingAsValidatorGroup() public {
+    validators.setValidatorGroup(delegator);
+    vm.startPrank(delegator);
+    vm.expectRevert("Validator groups cannot delegate votes.");
+    lockedGold.delegateGovernanceVotes(delegatee1, FixidityLib.newFixedFraction(1, 1).unwrap());
+  }
+
+  function test_ShouldRevertWhenIncorrectPercentAmountIsInserted() public {
+    vm.expectRevert("Delegate fraction must be less than or equal to 1");
+    lockedGold.delegateGovernanceVotes(delegatee1, FixidityLib.newFixedFraction(101, 100).unwrap());
+  }
+
+  function test_ShouldRevertWhenDelegatingVotesThatAreCurrentlyVotingForProposal_WhenDelegatorIsVotingInReferendum_WhenSomeGoldIsLocked()
+    public
+  {
+    uint256 value = 1000;
+
+    vm.startPrank(delegator);
+    lockedGold.lock.value(value)();
+    governance.setTotalVotes(delegator, 1);
+
+    vm.expectRevert("Cannot delegate votes that are voting in referendum");
+    lockedGold.delegateGovernanceVotes(delegatee1, FixidityLib.newFixedFraction(100, 100).unwrap());
+  }
+
+  function test_ShouldRevertWhenDelegatingVotesThatAreCurrentlyVotingForProposal2Delegatees_WhenDelegatorIsVotingInReferendum_WhenSomeGoldIsLocked()
+    public
+  {
+    uint256 value = 1000;
+
+    vm.startPrank(delegator);
+    lockedGold.lock.value(value)();
+    governance.setTotalVotes(delegator, 1);
+
+    lockedGold.delegateGovernanceVotes(delegatee1, FixidityLib.newFixedFraction(99, 100).unwrap());
+
+    vm.expectRevert("Cannot delegate votes that are voting in referendum");
+    lockedGold.delegateGovernanceVotes(delegatee2, FixidityLib.newFixedFraction(1, 100).unwrap());
+  }
+
+  function test_ShouldDelegateWhenVotingForLessThanRequestedForDelegatetion_WhenDelegatorIsVotingInReferendum_WhenSomeGoldIsLocked()
+    public
+  {
+    uint256 value = 1000;
+
+    vm.startPrank(delegator);
+    lockedGold.lock.value(value)();
+    governance.setTotalVotes(delegator, 1);
+
+    lockedGold.delegateGovernanceVotes(delegatee1, FixidityLib.newFixedFraction(99, 100).unwrap());
+
+    assertEq(lockedGold.getAccountTotalGovernanceVotingPower(delegator), 10);
+    assertEq(lockedGold.getAccountTotalGovernanceVotingPower(delegatee1), 990);
+  }
+
+  function test_ShouldRevertWhenDelegatingMoreThan100PercentInTwoStepsForDifferentDelegatees_WhenDelegatingToDelegatee1_WhenSomeGoldIsLocked()
+    public
+  {
+    uint256 value = 1000;
+
+    vm.prank(delegator);
+    lockedGold.lock.value(value)();
+    lockedGold.delegateGovernanceVotes(delegatee1, FixidityLib.newFixedFraction(10, 100).unwrap());
```

Comment:
I personally find it much easier to just write `0.1e24`.

Reply:
I created helper functions that hopefully make it more readable